### DISCUSSION TITLE
Checking the speed of algorithms

### DISCUSSION THREAD

**Discussion by viswans2132** [qpsolvers/qpsolvers#301] [Q&A]
Hey,

I would like to try different quadratic program algorithms with linear and nonlinear constraints and compare their performance and running time. Is there a way to find the optimization time using the solvers you've provided?

**Comment by stephane-caron** [comment]
You can always time solvers externally, but the timing will include both matrix build and solve times. This is what is currently done [qpbenchmark](https://github.com/qpsolvers/qpbenchmark/) in `qpbenchmark/utils.py`. Essentially:

```python
start_time = perf_counter()
solution = qpsolvers.solve_problem(problem, solver=solver)
runtime = perf_counter() - start_time
```

Separating build and solve times is open as an [area of improvement](https://github.com/qpsolvers/qpbenchmark/issues/12) for the benchmark. It should indeed be done here in qpsolvers.

I don't have the bandwidth to work on this in the next two months, but if a contributor steps in and proposes a PR I'd gladly help review and merge it :smiley:
